### DISCUSSION TITLE
Explosions

### DISCUSSION THREAD

**Discussion by para-dise** [Minestom/Minestom#1177] [Help] [answered]
Hi,

So I'm not really understanding how explosions work. I tried

```java
instance.explode(arrow.getPosition().blockX(), arrow.getPosition().blockY(), arrow.getPosition().blockZ(), 4.0f);
```

which throws

```java
java.lang.IllegalStateException: Tried to create an explosion with no explosion supplier
```

Although I really have no idea how to use this explosion supplier... if anyone could help that'd be appreciated!

**Comment by GreatWyrm** [comment, marked as answer]
You will need to do `instance.setExplosionSupplier(ExplosionSupplier);` before running the `instance.explode()` function.

ExplosionSupplier is an interface which you can read more about [here](https://javadoc.minestom.net/net/minestom/server/instance/ExplosionSupplier.html).

You can choose to create a class that implements the interface, or use a lambda instead.
